pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -330,6 +335,18 @@ def __init__(self, training: bool, device: Optional[Union[str, torch.device]] =
         self.device: Optional[Union[str, torch.device]] = device
         self.fx_validator = FxValidator()
 
+    @property
+    def result_metrics(self) -> List[ResultMetric]:
+        o = []
+        for v in self.values():
+            if isinstance(v, ResultMetric):
+                o.append(v)
+            elif isinstance(v, ResultCollection):
+                for _v in v.items():
+                    if isinstance(v, ResultMetric):
+                        o.append(_v)
```

Comment:
Can they be arbitrarily deep nested or is this the maximum depth?

Reply:
Done ! Good catch !